src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -1905,5 +1905,26 @@ void IslandWindow::UseMica(const bool newValue, const double /*titlebarOpacity*/
     // that version, this API will just return an error and do nothing silently.
 
     const int attribute = newValue ? DWMSBT_MAINWINDOW : DWMSBT_NONE;
-    std::ignore = DwmSetWindowAttribute(GetHandle(), DWMWA_SYSTEMBACKDROP_TYPE, &attribute, sizeof(attribute));
+    LOG_IF_FAILED(DwmSetWindowAttribute(GetHandle(), DWMWA_SYSTEMBACKDROP_TYPE, &attribute, sizeof(attribute)));
+}
+
+// Method Description:
+// - This method is called when the window receives the WM_NCCREATE message.
+// Return Value:
+// - The value returned from the window proc.
+[[nodiscard]] LRESULT IslandWindow::OnNcCreate(WPARAM wParam, LPARAM lParam) noexcept
+{
+    const auto ret = BaseWindow::OnNcCreate(wParam, lParam);
+    if (!ret)
+    {
+        return FALSE;
+    }
+
+    // This is a hack to make the window borders dark instead of light.
+    // It must be done before WM_NCPAINT so that the borders are rendered with
+    // the correct theme.
+    // For more information, see GH#6620.
+    LOG_IF_FAILED(TerminalTrySetDarkTheme(_window.get(), true));

```

Comment:
Can we guarantee that we'll always call `_updateTheme()`? If so, we could probably remove this.

Also, why do we pass in `true` for the second param here? I get that we should normally pass in the value from the theme and that's not ready yet, but should we even bother calling this function here then if we're waiting for the theme to load anyways?

Reply:
~oof that one might need to get yanked, good catch. The `true` is definitely wrong~


Er, no. The NonClientIslandWindow _always always_ needed this. Lemme double check what that looks like without it